server/schedule/operator_controller.go

```diff
@@ -91,7 +91,7 @@ func (oc *OperatorController) Dispatch(region *core.RegionInfo, source string) {
 	// Check existed operator.
 	if op := oc.GetOperator(region.GetID()); op != nil {
 		timeout := op.IsTimeout()
-		if step := op.Check(region); step != nil && !timeout {
+		if advanced, step := op.Check(region); step != nil && !timeout {
```

Comment:
There may be a problem here if the current step does not finish, should we send it again?